src/libcollections/vec.rs

```diff
@@ -1106,12 +1106,36 @@ impl<T: Clone> Vec<T> {
         let len = self.len();
 
         if new_len > len {
-            self.extend(repeat(value).take(new_len - len));
+            self.extend_elem(new_len - len, value);
         } else {
             self.truncate(new_len);
         }
     }
 
+    /// Extend the vector by `n` additional clones of `value`.
+    #[inline(always)]
```

Comment:
It's only used in two locations and I want it to inline into them (Vec::resize and from_elem). Do you think this is not going to work out that way? It's a private function.